### ISSUE TITLE
After snapshot operations fails, Rancher Desktop stays in a broken state

### ISSUE THREAD

**Issue by IsaSih** [rancher-sandbox/rancher-desktop#5809]
### Actual Behavior

If an error occurs while creating a snapshot, a pop up is displayed from the main window with a clickable button to `Show logs`. However, it fails to open the logs directory. After more investigation, it's possible to verify that the directory no longer exists and that a factory reset is performed but the VM is not restarted.

### Steps to Reproduce

1. Try to create a snapshot with `.DS_Store` in the snapshots directory
2. Try to open the logs directory by clicking `Show logs` from the error pop up that is displayed when the operation fails

### Result

<img width="1276" alt="Screenshot 2023-10-20 at 10 46 16 AM" src="https://github.com/rancher-sandbox/rancher-desktop/assets/70187881/98a76bee-f626-4a57-912f-adcad6b2edb4">

Factory reset removes the application resources and configuration, however the VM does not restart and the UI remains interactive.

```
ls $HOME/Library/Logs/rancher-desktop
ls: /Users/iguimaraes/Library/Logs/rancher-desktop: No such file or directory
```

```
ls $HOME/Library/Caches/rancher-desktop
ls: /Users/iguimaraes/Library/Caches/rancher-desktop: No such file or directory
```

```
ls "$HOME/Library/Application Support/rancher-desktop"

```
`Application Support/rancher-desktop` is empty

### Expected Behavior

When a snapshot operation fails, a factory reset must be performed and the VM must be restarted

### Rancher Desktop Version

1.10.0-455-g9e9a293a

### Rancher Desktop K8s Version

N/A

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

macOS

### Operating System / Build Version

macOS Ventura 13.5.2

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

None

**Comment by IsaSih** [comment]
Apparently, it breaks the functionality in Troubleshooting 
<img width="1270" alt="Screenshot 2023-10-20 at 5 50 19 PM" src="https://github.com/rancher-sandbox/rancher-desktop/assets/70187881/062cb57c-69bd-4e47-abb9-1f4318665f4f">

**Comment by jandubois** [comment]
I could not reproduce this issue because creating a `.DS_Store` file in the snapshot directory no longer creates an error.

I also believe that a failed "snapshot create" call will no longer perform a factory reset. I can't find the PR/commit, that changed this, but it was handled in the context of #5684.

I've talked with @adamkpickering and @ericpromislow and we agreed that it must be **impossible** for any snapshot-creation codepath to trigger a factory reset.

**Comment by jandubois** [comment]
I've been able to force a failure when creating a snapshot by mounting a filesystem at the snapshot directory location that doesn't have enough free space to hold a snapshot.

I've been using a "sparsebundle" on macOS; you should be able to do the same with a "tmpfs" on Linux. Not sure how to do this on Windows:

```console
$ SNAPSHOTS="$HOME/Library/Application Support/rancher-desktop/snapshots"
$ rm -rf "$SNAPSHOTS"
$ mkdir "$SNAPSHOTS"
$ hdiutil attach -mountpoint "$SNAPSHOTS" "$PWD/snapshots.sparsebundle"
/dev/disk11         	GUID_partition_scheme
/dev/disk11s1       	Apple_HFS                      	/Users/jan/Library/Application Support/rancher-desktop/snapshots
$ hdiutil attach -mountpoint "$SNAPSHOTS" "$PWD/snapshots.sparsebundle"
/dev/disk11         	GUID_partition_scheme
/dev/disk11s1       	Apple_HFS                      	/Users/jan/Library/Application Support/rancher-desktop/snapshots
```

Creating the snapshot from the UI will shut down the VM and then display this error dialog:

<img width="654" alt="CleanShot 2023-10-23 at 19 07 03@2x" src="https://github.com/rancher-sandbox/rancher-desktop/assets/78947/8c007726-6a24-44c4-a6bf-dd78bc6eaf5f">

It does **not** cause a factory reset, and the log files exist. The link to the log files works, as does the link in the troubleshooting page.

However, the backend is not restarting at this point. Even after closing the dialog, the backend is still not restarting.

After terminating Rancher Desktop you can unmount the sparsebundle with

```console
$ hdiutil detach "$SNAPSHOTS"
"disk11" ejected.
```
@adamkpickering I believe it is the responsibility of the `rdctl` code to restart the backend after reporting the error to the caller.

**Comment by jandubois** [comment]
I just realized that this is now a duplicate of #5764 and should be fixed by #5816.
